File: backend/models/cart_product.go
package models

import "time"

type CartProduct struct {
	ID        int64 `gorm:"primaryKey;autoIncrement"`
	CartID    int64
	Cart      Cart `gorm:"foreignKey:CartID"`
	ProductID int64
	Product   Product `gorm:"foreignKey:ProductID"`
	Quantity  int     `gorm:"not null"`
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/types/cart-product.ts
import { Product } from "./product";

export interface CartProduct {
  id: number;
  cart_id: number;
  product_id: number;
  product: Product;
  quantity: number;
  created_at: string;
  updated_at: string;
}

export interface Cart {
  id: number;
  user_id: number;
  items: CartProduct[];
}

export interface CartItem {
  id: number;
  quantity: number;
  product: {
    id: number;
    title: string;
    thumbnail: string;
    slug: string;
    description: string;
    price: number;
    stock: number;
    category: string;
  };
}

export interface AddToCartRequest {
  product_id: number;
  quantity: number;
}

export interface UpdateCartQuantityRequest {
  product_id: number;
  quantity: number;
}